src/common/checksum.c

```diff
@@ -299,17 +357,42 @@ daos_csummer_finish(struct daos_csummer *obj)
 }
 
 bool
-daos_csummer_compare(struct daos_csummer *obj, daos_csum_buf_t *a,
-		     daos_csum_buf_t *b)
+daos_csummer_dcb_compare(struct daos_csummer *obj, daos_csum_buf_t *a,
+			 daos_csum_buf_t *b)
 {
-	uint32_t a_len = a->cs_len * a->cs_nr;
-	uint32_t b_len = b->cs_len * b->cs_nr;
+	uint32_t	a_len = a->cs_len * a->cs_nr;
+	uint32_t	b_len = b->cs_len * b->cs_nr;
```

Comment:
a->cs_nr and b->cs_nr are equal or can be different

Reply:
"Should" be the same, but if the data the checksums were derived from is different size, than could be different.